spec/rester/utils/rspec_utils_spec.rb

```diff
@@ -0,0 +1,274 @@
+              {
+                this: :that,
+                some_hash: {
+                  some_array: [1, 2, 3]
+                }
+              }
+            }
+
+            it 'should raise an error' do
+              expect { subject }.to raise_error Errors::StubError, "Stub[:some_hash][:some_array]=[1, 2, 3] doesn't match Subject[:some_hash][:some_array]=nil"
+            end
+          end # with missing Array in Hash
+
+          context 'with missing hash key' do
+            let(:subj) {
+              {
+                this: :that,
+                some_hash: {
+                  a_different_key: :some_value
+                }
+              }
+            }
+            let(:stub) {
+              {
+                this: :that,
+                some_hash: {
+                  some_key: :some_value
+                }
+              }
+            }
+
+            it 'should raise an error' do
+              expect { subject }.to raise_error Errors::StubError, "Stub[:some_hash][:some_key]=:some_value doesn't match Subject[:some_hash][:some_key]=nil"
+            end
+          end # with missing hash key
+
+          context 'with missing array element' do
+            let(:subj) {
+              {
+                this: :that,
+                some_array: [1, 2]
+              }
+            }
+            let(:stub) {
+              {
+                this: :that,
+                some_array: [1, 2, 3]
+              }
+            }
+
+            it 'should raise an error' do
+              expect { subject }.to raise_error Errors::StubError, "Stub[:some_array][2]=3 doesn't match Subject[:some_array][2]=nil"
+            end
+          end # with missing array element
+
+          context 'with different hash value' do
+            let(:subj) {
+              {
+                this: :that,
+                some_hash: {
+                  some_key: :a_different_value
+                }
+              }
+            }
+            let(:stub) {
+              {
+                this: :that,
+                some_hash: {
+                  some_key: :some_value
+                }
+              }
+            }
+
+            it 'should raise an error' do
+              expect { subject }.to raise_error Errors::StubError, "Stub[:some_hash][:some_key]=:some_value doesn't match Subject[:some_hash][:some_key]=:a_different_value"
+            end
+          end # with different hash value
+
+          context 'with different array element' do
+            let(:subj) {
+              {
+                this: :that,
+                some_array: [1, 2, 4]
+              }
+            }
+            let(:stub) {
+              {
+                this: :that,
+                some_array: [1, 2, 3]
+              }
+            }
+
+            it 'should raise an error' do
+              expect { subject }.to raise_error Errors::StubError, "Stub[:some_array][2]=3 doesn't match Subject[:some_array][2]=4"
+            end
```

Comment:
Thoughts on renaming `Subject` to `Response`?  That may make it more clear that we're comparing the stub with the response from the service.